GeneratorPgn.py

```diff
@@ -0,0 +1,318 @@
+        fen_dropped = fen[:-1]
+        fen_dropped.extend(fen[-1].split())
+        board_position_str = fen_dropped[:8]
+        next_mover_str = fen_dropped[8]
+        castling_str = fen_dropped[9]
+        en_passant_str = fen_dropped[10]
+        half_moves_str = fen_dropped[11]
+        move_idx_str = fen_dropped[12]
+ 
+        board_position = np.zeros((12, 8, 8,), dtype=np.float32)
+        for r_idx, row in enumerate(board_position_str):
+            column = 0
+            for piece in row:
+                try:
+                    p = pieces[piece]
+                    board_position[p, column, r_idx] = 1.
+                    column += 1
+                except KeyError:
+                    column += int(piece)
+ 
+        next_mover = -1 if next_mover_str == 'b' else 1
+        castling = np.zeros(4)
+        for figure in castling_str:
+            if figure == 'K':
+                castling[0] = 1
+            elif figure == 'Q':
+                castling[1] = 1
+            elif figure == 'k':
+                castling[2] = 1
+            elif figure == 'q':
+                castling[3] = 1
+        en_passant = np.zeros((1, 8, 8,), dtype=np.float32)
+        if en_passant_str != '-':
+            en_passant[0, first_cords[en_passant_str[0]], second_cords[en_passant_str[1]]] = 1.
+        half_moves = int(half_moves_str)
+        move_idx = int(move_idx_str)
+        return {'board': board_position,
+                'castling': castling,
+                'en_passant': en_passant,
+ 
+                'player': next_mover,
+                'half_move': half_moves, 'full_move': move_idx,
+                'fen': raw_fen}
+ 
+    def preprocess_batch(self, batch):
+        """
+        Applies `preprocess_fen` for a batch of data.
+        :param batch: batch to preprocess
+        :return: dictionary of preprocessed tensors
+        """
+        boards = []
+        players = []
+        castlings = []
+        en_passants = []
+        q_values = []
+        half_moves = []
+        full_moves = []
+        fens = []
+        for fen in batch:
+            data = self.preprocess_fen(fen)
+            boards.append(data['board'])
+            players.append(data['player'])
+            castlings.append(data['castling'])
+            en_passants.append(data['en_passant'])
+            half_moves.append(data['half_move'])
+            full_moves.append(data['full_move'])
+            fens.append(data['fen'])
+ 
+        return {'boards': boards,
+                'players': players,
+                'castlings': castlings,
+                'en_passants': en_passants,
+                'half_moves': half_moves,
+                'full_moves': full_moves}
+ 
+    def __iter__(self):
+        """
+        Like `Iterator.__iter__()`, but applies preprocessing to yielded data.
+        :return: yields preprocessed batches of samples
+        """
+        for batch in self.iterator:
+            #borr=chess.Board().set_fen(batch)

```

Comment:
will this code be needed? If no, nuke it - commented code is a bad practice because it quickly becomes unusable